syntax = "proto3";

import "google/protobuf/timestamp.proto";

message UserByIdRequest {
  string user_id = 1;
}

message UserResponse {
  int64 status_code = 1;
  optional User user = 2;
  optional string message = 3;
}

message UsersResponse {
  int64 status_code = 1;
  optional ListOfUser users = 2;
  optional string message = 3;
}

message ListOfUser {
  repeated User users = 1;
}

enum UserType {
    USER = 0;
    ADMIN = 1;
}

message User {
  string id = 1;
  string username = 2;
  string email = 3;
  google.protobuf.Timestamp created_at = 4;
  optional google.protobuf.Timestamp suspended_at = 5;
  UserType type = 6;
}

message UsersByIdRequest {
  repeated string id = 1;
}
